control-plane/agents/src/bin/core/controller/scheduling/volume_policy/simple.rs

```diff
@@ -35,46 +39,95 @@ impl ResourcePolicy<AddVolumeReplica> for SimplePolicy {
+    fn vg_replica_count() -> SortCriteria {
+        SortCriteria::new(
+            Criteria::new("vg_replica_count", VG_REPL_COUNT_WEIGHT),
+            ValueGrading::Lower,
+            |pool_item| pool_item.vg_replica_count().into(),
+        )
+    }
+    /// SortCriteria for number of total replicas on pool only for volume group.
+    fn vg_total_replica_count() -> SortCriteria {
+        SortCriteria::new(
+            Criteria::new("vg_total_replica_count", VG_TOTAL_REPLICA_COUNT_WEIGHT),
+            ValueGrading::Lower,
+            |pool_item| pool_item.len().into(),
+        )
+    }
+    /// SortCriteria for number of total replicas on pool.
+    fn non_vg_total_replica_count() -> SortCriteria {
+        SortCriteria::new(
+            Criteria::new("non_vg_total_replica_count", TOTAL_REPLICA_COUNT_WEIGHT),
+            ValueGrading::Lower,
+            |item| item.len().into(),
+        )
+    }
+    /// SortCriteria for over commitment on pool.
+    fn over_commitment() -> SortCriteria {
+        SortCriteria::new(
+            Criteria::new("over_commitment", OVER_COMMIT_WEIGHT),
+            ValueGrading::Lower,
+            |item| item.pool().over_commitment().into(),
+        )
+    }
     /// Sort pools using weights between:
-    /// 1. number of replicas (N_REPL_WEIGHT %)
+    /// 1. number of replicas or number of replicas of a vg (N_REPL_WEIGHT %)
     /// 2. free space         (FREE_SPACE_WEIGHT %)
     /// 3. overcommitment     (OVER_COMMIT_WEIGHT %)
-    pub(crate) fn sort_by_weights(a: &PoolItem, b: &PoolItem) -> std::cmp::Ordering {
-        const N_REPL_WEIGHT: Ranged = Ranged::new_const(25);
-        const FREE_SPACE_WEIGHT: Ranged = Ranged::new_const(40);
-        const OVER_COMMIT_WEIGHT: Ranged = Ranged::new_const(35);
-
-        let n_replicas = Criteria::new("n_replicas", N_REPL_WEIGHT);
-        let free_space = Criteria::new("free_space", FREE_SPACE_WEIGHT);
-        let over_commit = Criteria::new("over_commit", OVER_COMMIT_WEIGHT);
-
-        let (score_a, score_b) = WeightedScore::dual_values()
-            .weigh(n_replicas, ValueGrading::Lower, a.len(), b.len())
-            .weigh(
-                free_space,
-                ValueGrading::Higher,
-                a.pool().free_space(),
-                b.pool().free_space(),
-            )
-            .weigh(
-                over_commit,
-                ValueGrading::Lower,
-                a.pool().over_commitment(),
-                b.pool().over_commitment(),
-            )
-            .score()
-            .unwrap();
-
-        score_b.cmp(&score_a)
+    pub(crate) fn sort_by_weights(
+        request: &GetSuitablePoolsContext,
+        a: &PoolItem,
+        b: &PoolItem,
+    ) -> std::cmp::Ordering {
+        match a.pool.state().status.partial_cmp(&b.pool().state().status) {
+            Some(Ordering::Greater) => Ordering::Greater,
+            Some(Ordering::Less) => Ordering::Less,
+            Some(Ordering::Equal) => {
```

Comment:
```suggestion
            None | Some(Ordering::Equal) => {
```
Makes sense to groups this together I think.

Reply:
Indeed. Done.